services/src/main/java/org/keycloak/authentication/authenticators/client/ClientIdAndSecretAuthenticator.java

```diff
@@ -56,6 +56,7 @@ public void authenticateClient(ClientAuthenticationFlowContext context) {
         String clientSecret = null;
 
         String authorizationHeader = context.getHttpRequest().getHttpHeaders().getRequestHeaders().getFirst(HttpHeaders.AUTHORIZATION);
+        String requestPath = context.getHttpRequest().getUri().getPath();
```

Comment:
I believe that if we replace the 
`String requestPath = context.getHttpRequest().getUri().getPath();`
with 
`String requestPath = context.getSession().getContext().getUri().getPath();`
will resolve the problem with the failing tests

Reply:
@laskasn done

Reply:
seems to be ok so far. Let's wait for it to finish.
But let's see if such a modification is spec-compliant. 
TMHO, i believe that we should contact the team responsible for oauch and inform them about the spec's description. Maybe they know something more, that we are missing right now. 
What do you think guys?